### DISCUSSION TITLE
Email Wallet on behalf of user

### DISCUSSION THREAD

**Discussion by irfanonk** [WalletConnect/walletconnect-monorepo#5541] [Developer Support]
Hello,

I have already integrated your wallet service into my app, and I have a specific requirement for users who have already logged in with their email on my platform. I want to simplify the wallet creation process for users unfamiliar with web3. Here's what I need:

1. Is it possible to create a wallet on behalf of the user in the background using their login email without showing the modal?

2. If not, can the email be pre-filled and the modal jump directly to the OTP screen, skipping any intermediate steps?

**Comment by arein** [comment]
@irfanonk very cool! where can we see your app?

> Is it possible to create a wallet on behalf of the user in the background using their login email without showing the modal?

We don't support this right now or in the near future

> If not, can the email be pre-filled and the modal jump directly to the OTP screen, skipping any intermediate steps?

We were discussing a `hook` to allow to do that.

@Sam-Newman for prioritization

**Reply by irfanonk** [reply]
thats great if you provide this kind of feature that would be appriciated.
here is the url : [udao](https://app.udao.org/)

**Reply by irfanonk** [reply]
Regarding option 2, you mentioned that you were discussing a hook to allow pre-filling the email and jumping directly to the OTP screen. Could you please provide more details on this?

1. Is this feature currently available or in development?
2. If in development, do you have an estimated timeline for when it might be released?
3. Are there any alternative approaches we can implement in the meantime? 
4. is it possible and feasible to fill the email input and click to proceed button by manipulating DOM

**Reply by arein** [reply]
4) hypothetically yes but that is not a stable way of doing it

I checked out your website - I recommend you enable email/social login so users can connect accounts for now. Once we have the functionality to either pregenerate wallets or to prefill the email address we will let you know.

It's not in active development but we'd be happy to build this with you.

**Reply by irfanonk** [reply]
i would be happy to help for sure
